dali/operators/sequence/optical_flow/optical_flow.cc

```diff
@@ -81,6 +82,40 @@ constexpr int kNInputDims = 4;
 
 template<>
 void OpticalFlow<GPUBackend>::RunImpl(Workspace<GPUBackend> &ws) {
+  // This is a workaround for an issue with nvcuvid in drivers >460 where concurrent
+  // use on default context and non-default streams may lead to memory corruption.
+  // TODO(michalz): add an upper bound when the problem is fixed
+  cudaStream_t of_stream = ws.stream();
+#if NVML_ENABLED
+  {
+    static float driver_version = []()->float {
+      char version[80];
+      CUDA_CALL(nvmlInitChecked());
+      CUDA_CALL(nvmlSystemGetDriverVersion(version, sizeof version));
+
+      return std::stof(version);
+    }();
+    if (driver_version > 460)
+      of_stream = 0;
+  }
+#else
+  {
+    int driver_cuda_version = 0;
+    CUDA_CALL(cuDriverGetVersion(&driver_cuda_version));
+    if (driver_cuda_version >= 11030)
+      of_stream = 0;
```

Comment:
Any reason to provide both this and the NVML implementation?

Reply:
With NVML we can distinguish driver versions, not CUDA versions - that's finer granularity, we can possibly pick up fixes earlier.

Reply:
How this work with 450.x driver and compatibility layer enabled? I think `cuDriverGetVersion` will report 11030.

Reply:
I have no idea, but that's for bare-metal builds without NVML, so I wouldn't sweat it too much.

Reply:
I think it will work as I said. 450.x will report 11030. Far from perfect TBH.